docs/basics.md

```diff
@@ -157,6 +157,153 @@ other things yourself, if needed.
 
 [Read more about customising `urql` in the "Extending & Experimenting" section.](extending-and-experimenting.md)
 
+## Server-side rendering
+
+Server-side rendering is a common method to reduce the time it takes for
+a user to see a React page's content. Typically this is implemented using
+the [`react-dom/server` package](https://reactjs.org/docs/react-dom-server.html).
+
+`urql` can be set up to fetch data on the server and rehydrate this data
+on the client, so that the user's browser does not need to refetch it and
+can seemlessly rehydrate your React page.
+
+There's two parts in `urql` that enable server-side rendering to work:
+
+- The `Client` has a `suspense` option, which enables support for React's
+  experimental Suspense API for data fetching, which allows us to prefetch
+  data before calling `renderToString` or `renderToNodeStream`.
+- The `ssrExchange` which is a small operation cache that works together
+  with Suspense to save data on the server and rehydrate it on the client.
+
+Since Suspense is still an experimental API there's no official way to use
+it to prefetch data on the server-side. For this reason we have a companion
+library, [`react-ssr-prepass`](it currently doesn't hav), which can be used to run a "prepass"
+that fetches all suspended data it finds in a React element tree.
+
+### Setting up the `Client`
+
+When you set up the `Client` for server-side rendering, on the server
+you will need to set `suspense` to `true` and on the client to `false`,
+
+```js
+import { Client } from 'urql';
+
+const client = new Client({
+  suspense: !process.browser,
+  // ...
+});
+```
+
+You can often achieve this with `process.browser` in most settings if
+you're using a single universal file to create a client on the server
+and on the client.
+
+Next up, the `ssrExchange` needs to be set up. It's a factory, since
+it has some methods for extracting and rehydrating data.
+
+```js
+import {
+  Client,
+  dedupExchange,
+  cacheExchange,
+  fetchExchange,
+  ssrExchange,
+} from 'urql';
+
+const ssrCache = ssrExchange();
+
+const client = new Client({
+  exchanges: [
+    dedupExchange,
+    cacheExchange,
+    ssrCache, // Added here after calling ssrExchange
+    fetchExchange,
+  ],
+  // ...
+  suspense: !process.browser,
+});
+```
+
+The exchange from `ssrExchange()` should be added after the default
+or your custom `cacheExchange` and before any asynchronous exchanges
+like the `fetchExchange`.
+
+### Prefetching on the server
+
+In your request handler on the server-side, you'll have to add some
+code handling suspense. Typically this is done using a "prepass" that
+walks your element tree and awaits suspended promises.
+
+So for suspense to now be executed on the server you may install
+`react-ssr-prepass` (which has a peer dependency on `react-is`)
```

Comment:
It would be good to have short paragraph comparing and contrasting this `react-ssr-prepass` to `getDataFromTree`, so people understand its role